sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/RequestResponseChannelCache.java

```diff
@@ -0,0 +1,279 @@
+    private volatile boolean terminated;
+    // Note: The only reason to have below 'currentChannel' is to close the cached RequestResponseChannel internally
+    // upon 'RequestResponseChannelCache' termination (via dispose()). We must never expose 'currentChannel' variable to
+    // any dependent type; instead, the dependent type must acquire RequestResponseChannel only through the cache route,
+    // i.e., by subscribing to 'createOrGetCachedChannel' via 'get()' getter.
+    private volatile RequestResponseChannel currentChannel;
+
+    RequestResponseChannelCache(ReactorConnection connection, String entityPath, String sessionName, String linksName,
+        AmqpRetryPolicy retryPolicy) {
+        Objects.requireNonNull(connection, "'connection' cannot be null.");
+        Objects.requireNonNull(entityPath, "'entityPath' cannot be null.");
+        Objects.requireNonNull(sessionName, "'sessionName' cannot be null.");
+        Objects.requireNonNull(linksName, "'linksName' cannot be null.");
+        Objects.requireNonNull(retryPolicy, "'retryPolicy' cannot be null.");
+
+        // for correlation purpose, the cache and any RequestResponseChannel it caches uses the same loggingContext. E.g,
+        // { "connectionId": "MF_0f4c2e_1680070221023" "linkName": 'cbs' or '{entityPath}-mgmt' (e.g. 'q0-mgmt', 'tx-mgmt') }
+        final Map<String, Object> loggingContext = new HashMap<>(2);
+        loggingContext.put(CONNECTION_ID_KEY, connection.getId());
+        loggingContext.put(LINK_NAME_KEY, linksName);
+        this.logger = new ClientLogger(RequestResponseChannelCache.class, loggingContext);
+
+        this.connection = connection;
+        this.activationTimeout = retryPolicy.getRetryOptions().getTryTimeout();
+
+        final Mono<RequestResponseChannel> newChannel = Mono.defer(() -> {
+            final RecoveryTerminatedException terminatedError = checkRecoveryTerminated("new-channel");
+            if (terminatedError != null) {
+                return Mono.error(terminatedError);
+            }
+            return connection.newRequestResponseChannel(sessionName, linksName, entityPath);
+        });
+
+        this.createOrGetCachedChannel = newChannel
+            .flatMap(c -> {
+                logger.atInfo()
+                    .log("Waiting for channel to active.");
+
+                final Mono<RequestResponseChannel> awaitToActive = c.getEndpointStates()
+                    .filter(s -> s == AmqpEndpointState.ACTIVE)
+                    .next()
+                    .switchIfEmpty(Mono.error(() -> new AmqpException(true, "Channel completed without being active.", null)))
+                    .then(Mono.just(c))
+                    .timeout(activationTimeout, Mono.defer(() -> {
+                        final String timeoutMessage = String.format("The channel activation wait timed-out (%s).", activationTimeout);
+                        logger.atInfo().log(timeoutMessage + " Closing channel.");
```

Comment:
We put it there identify the case if the later `closeAsync` is stuck blocking retry, unsure if that ever happens but just in case there some corner cases (thoughts?)